clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -283,6 +290,10 @@ public synchronized void removeListener(Listener listener) {
         this.listeners.remove(listener);
     }
 
+    public void setClusterResourceListeners(ClusterResourceListeners clusterResourceListeners) {
```

Comment:
Do we actually expect ClusterResourceListeners to be updated? if not, would it be better to pass in ClusterResourceListeners in the constructor?